source/mainnet/smart-contracts/guides/contract-dev-guides.rst

```diff
@@ -26,3 +26,4 @@ Once you are familiar with smart contracts, it is a good idea to read the :ref:`
    upgradeable-contract
    json-params
    migrate-contracts
+   factory-pattern
```

Comment:
This is not really written as a tutorial, so I'd rather not move it there. I'll add a link from the Ethereum FAQ, but since there aren't other guides under it, I think it's reasonable to keep it in the guides section (for now).